src/templates/React InstantSearch widget/tsconfig.declaration.json

```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "./tsconfig",
+  "compilerOptions": {
+    "outDir": "dist",
+    "noEmit": false,
+    "declaration": true,
+    "emitDeclarationOnly": true
+  },
+  "exclude": ["./src/main.tsx", "**/__tests__"]
```

Comment:
`main.tsx` is just for example, so we do not emit types for it.

Reply:
Shouldn't we rather have one folder for the lib, and one for the demo?